codalab/server/bundle_manager.py

```diff
@@ -409,20 +408,20 @@ def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run, user_
                     worker['exit_after_num_runs'] -= 1
                     break
 
-        # To avoid the potential race condition between bundle manager's dispatch frequency and
-        # worker's checkin frequency, update the column "exit_after_num_runs" in worker table
-        # before bundle manager's next scheduling loop
-        for worker in workers_list:
-            # Update workers that have "exit_after_num_runs" manually set from CLI.
-            if (
-                worker['exit_after_num_runs']
-                < workers._workers[worker['worker_id']]['exit_after_num_runs']
-            ):
-                self._worker_model.update_workers(
-                    worker["user_id"],
-                    worker['worker_id'],
-                    {'exit_after_num_runs': worker['exit_after_num_runs']},
-                )
+            # To avoid the potential race condition between bundle manager's dispatch frequency and
```

Comment:
@nelson-liu If I'm not mistaken, I think the old indentation was correct. The new indentation will update the worker table in the database for every staged bundle. Workers with a value of 0 for `exit_after_num_runs` will get filtered out in line 395.

Reply:
Hmm, how was this code working before, then? as-is, the value of `workers_list` depends on what the workers for the _last_ bundle are (i.e., it's bundle-order dependent). So, you could have different results depending on the bundle order, which doesn't seem ideal...thoughts?